### PR TITLE
Related Group Module

### PR DESCRIPTION
**Pull request by SunCatMC**
## Purpose
Add ability to create and manage groups of related tickets

## Approach

- Creating a module that launches submodules which are using commands inside arisa's comment to add, remove, and keep updated links of related group
- Creating commands to manage the arisa comment and via that the submodules

#### Checklist
- [ ] Included tests
- [ ] Updated documentation in [README](https://github.com/mojira/arisa-kt/blob/master/README.md) and [wiki](https://github.com/mojira/arisa-kt/wiki)
- [ ] Tested in MCTEST-xxx

### CODE REVIEW

**Comment by SunCatMC** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by SunCatMC** [comment]
recheckcla

File: src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            val arisaComments = comments
+                .filter(::isStaffRestricted)
+                .filter(::userIsArisa)
+                .filter(::isProbablyARelatedGroup)
+            assertNotEmpty(arisaComments).bind()
+
+            val isError = arisaComments.any {
+                !it.body.isNullOrBlank()
```
Comment (urielsalis) [inline src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt:26]: there is a isNotNull 👀

Comment (SunCatMC) [reply src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt:26]: no? there's only isNotBlank and isNotEmpty, neither of which check for null (based on syntax error)

Comment (SunCatMC) [reply src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt:25]: and regardless, I need both checks, and they are conveniently combined into one

Comment (SunCatMC) [reply src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt:25]: or actually no, don't need both - assert works with empty or null result, and I just need to pass the syntax

Comment (SunCatMC) [reply src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt:25]:
```suggestion
                it.body != null
```
I guess this what you were thinking about?

Comment (urielsalis) [reply src/main/kotlin/io/github/mojira/arisa/modules/RelatedGroupModule.kt:25]: nvm, brainfart :P

**Comment by SunCatMC** [comment]
closing it as I'm gonna rewrite it as a comand instead (in #452 ability to access other tickets from a ticket is added, making it possible to implement a simple "link group of tickets" instead of a complicated module)

**Pull request closed without merging**